src/databricks/labs/ucx/source_code/linters/python_ast.py

```diff
@@ -118,6 +123,140 @@ def first_statement(self):
+
+    def nodes_between(self, first_line: int, last_line: int) -> list[NodeNG]:
+        if not isinstance(self.node, Module):
+            raise NotImplementedError(f"Can't extract nodes from {type(self.node).__name__}")
+        self_module: Module = cast(Module, self.node)
+        nodes: list[NodeNG] = []
+        for node in self_module.body:
+            if node.lineno < first_line:
+                continue
+            if node.lineno > last_line:
+                break
+            nodes.append(node)
+        return nodes
+
+    def globals_between(self, first_line: int, last_line: int) -> dict[str, list[NodeNG]]:
+        if not isinstance(self.node, Module):
+            raise NotImplementedError(f"Can't extract globals from {type(self.node).__name__}")
+        self_module: Module = cast(Module, self.node)
+        globs: dict[str, list[NodeNG]] = {}
+        for key, nodes in self_module.globals.items():
+            nodes_in_scope: list[NodeNG] = []
+            for node in nodes:
+                if node.lineno < first_line or node.lineno > last_line:
+                    continue
+                nodes_in_scope.append(node)
+            if len(nodes_in_scope) > 0:
+                globs[key] = nodes_in_scope
+        return globs
+
+    def line_count(self):
+        if not isinstance(self.node, Module):
+            raise NotImplementedError(f"Can't count lines from {type(self.node).__name__}")
+        self_module: Module = cast(Module, self.node)
+        nodes_count = len(self_module.body)
+        if nodes_count == 0:
+            return 0
+        return 1 + self_module.body[nodes_count - 1].lineno - self_module.body[0].lineno
+
+    def renumber(self, start: int) -> Tree:
+        assert start != 0
+        if not isinstance(self.node, Module):
+            raise NotImplementedError(f"Can't renumber {type(self.node).__name__}")
+        root: Module = self.node
+        # for now renumber in place to avoid the complexity of rebuilding the tree with clones
+
+        def renumber_node(node: NodeNG, offset: int):
+            for child in node.get_children():
+                renumber_node(child, offset + child.lineno - node.lineno)
+            if node.end_lineno:
+                node.end_lineno = node.end_lineno + offset
+                node.lineno = node.lineno + offset
+
+        nodes = root.body
+        if start > 0:
+            for node in nodes:
+                offset = start - node.lineno
+                renumber_node(node, offset)
+                num_lines = 1 + (node.end_lineno - node.lineno if node.end_lineno else 0)
+                start = start + num_lines
+        else:
+            for node in reversed(nodes):
+                offset = start - node.lineno
+                renumber_node(node, offset)
+                num_lines = 1 + (node.end_lineno - node.lineno if node.end_lineno else 0)
+                start = start - num_lines
+
+        return self
+
+
+class TreeHelper(ABC):
```

Comment:
don't introduce new class and make only modifications to `append_tree` / `append_globals` / `append_nodes` / `is_from_module` bodies

Reply:
This is to avoid linting errors with public methods > 20